paint/data/stac_client.py

```diff
@@ -0,0 +1,560 @@
+                    mappings.SAVE_CALIBRATION,
+                    start_date,
+                    end_date,
+                    filtered_calibration_keys,
+                )
+
+            # Download deflectometry data.
+            if get_deflectometry:
+                deflectometry_id = (
+                    mappings.DEFLECTOMETRY_COLLECTION_ID
+                    % heliostat_catalog.id.split("-")[0]
+                )
+                self.download_heliostat_data(
+                    heliostat_catalog,
+                    deflectometry_id,
+                    mappings.SAVE_DEFLECTOMETRY,
+                    start_date,
+                    end_date,
+                )
+
+            # Download properties data.
+            if get_properties:
+                properties_id = (
+                    mappings.HELIOSTAT_PROPERTIES_COLLECTION_ID
+                    % heliostat_catalog.id.split("-")[0]
+                )
+                self.download_heliostat_data(
+                    heliostat_catalog,
+                    properties_id,
+                    mappings.SAVE_PROPERTIES,
+                    start_date,
+                    end_date,
+                )
+
+    def download_weather_assets(
+        self, weather_item, include_juelich: bool, include_dwd: bool
+    ):
+        """
+        Download weather assets for Jülich and/or DWD depending on flags.
+
+        Parameters
+        ----------
+        weather_item : pystac.item.Item
+            Weather item containing assets.
+        include_juelich : bool
+            Indicating whether to include Jülich data.
+        include_dwd : bool
+            Indicating whether to include DWD data.
+        """
+        if "juelich" in weather_item.id and include_juelich:
+            log.info(
+                f"Processing and downloading Jülich weather data item {weather_item.id}"
+            )
+        elif "dwd" in weather_item.id and include_dwd:
+            log.info(
+                f"Processing and downloading DWD weather data item {weather_item.id}"
+            )
+        else:
+            return  # If item does not match filters, skip
+
+        # Download the assets
+        for asset in weather_item.assets.values():
+            url = asset.href
+            file_end = asset.href.split("/")[-1]
+            file_name = self.output_dir / mappings.SAVE_WEATHER / file_end
+            file_name.parent.mkdir(parents=True, exist_ok=True)
+            self.download_file(url=url, file_name=file_name)
+
+    def get_weather_data(
+        self,
+        data_sources: Optional[list[str]] = None,
+        start_date: Optional[datetime] = None,
+        end_date: Optional[datetime] = None,
+    ) -> None:
+        """
+        Get weather data.
+
+        Parameters
+        ----------
+        data_sources : list[str], optional
+            List of weather sources to include. Must be from: `DWD, Jülich`. If no source is provided data form all
```

Comment:
```suggestion
            List of weather sources to include. Must be from: `DWD`, `Jülich`. If no source is provided data form all
```
Is it really with the "ü" 🙈?